uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -559,71 +414,87 @@ define([
      * {Promise}
      */
     function initContact () {
-      if (this.canManage) {
-        return Contact.find(this.request.contact_id)
+      if (vm.canManage) {
+        return Contact.find(vm.request.contact_id)
           .then(function (contact) {
-            this.contactName = contact.display_name;
-          }.bind(this));
+            vm.contactName = contact.display_name;
+          });
       }
 
       return $q.resolve();
     }
 
     /**
-     * Initialize from and to dates and day types.
-     * It will also set the day types.
-     *
-     * @return {Promise}
-     */
-    function initDates () {
-      if (!this.isMode('create')) {
-        var attributes = this.request.attributes();
-
-        this.uiOptions.fromDate = this._convertDateFormatFromServer(this.request.from_date);
-
-        return this.loadAbsencePeriodDatesTypes(this.uiOptions.fromDate, 'from')
-          .then(function () {
-            // to_date and type has been reset in above call so reinitialize from clone
-            this.request.to_date = attributes.to_date;
-            this.request.to_date_type = attributes.to_date_type;
-            this.uiOptions.toDate = this._convertDateFormatFromServer(this.request.to_date);
-            return this.loadAbsencePeriodDatesTypes(this.uiOptions.toDate, 'to');
-          }.bind(this));
-      } else {
-        return $q.resolve();
-      }
-    }
-
-    /**
      * Initialises listeners
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::requestObjectUpdated', setInitialAttributes.bind(this)),
-        $rootScope.$on('LeaveRequestPopup::childComponent::register', function () { childComponentsCount++; })
+        $rootScope.$on('LeaveRequestPopup::requestObjectUpdated', setInitialAttributes),
+        $rootScope.$on('LeaveRequestPopup::handleError', function (__, errors) {
```

Comment:
I think this can be one line, the previous syntax was better I think. It saves 2 lines and shows the simplicity of code.